lib/types/svg.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+var fs        = require('fs');
+var Promise   = require('bluebird');
+var pread     = Promise.promisify(fs.read);
+var preadFile = Promise.promisify(fs.readFile);
+var utils     = require('../utils');
+
+var regExps = {
+  rootTag: /<svg\s[^>]+>/,
+  width: /\bwidth="([^"%]+)"/,
+  height: /\bheight="([^"%]+)"/,
+  viewBox: /\bviewBox="([^"]+)"/
+};
+
+function isSVG (buffer) {
+  return ('<' === utils.ascii(buffer, 0, 1));
+}
+
+function measure (path, fd) {
+  return preadFile(path)
```

Comment:
We don’t really need to read the whole file — the opening `<svg>` would be enough.